### PR TITLE
upgrades: Update docs

### LINKED ISSUE #51172: Improve documentation on how to resolve schema drift

### PR DESCRIPTION
**Pull request by efritz**
Update docs related to MVU after Q2 work has completed. Fixes #51172 and #51170.

## Test plan

Docs update :)

### CODE REVIEW

**Review by Numbers88s — APPROVED** [review]

**Comment by courier-new** [comment]
👋 Just an FYI to save you some time because I also tried to do this earlier haha, backports aren't necessary for docs-only changes since `docs.sourcegraph.com` just deploys the latest on `main` and customers don't host their own versioned docs anymore. 😄

File: doc/admin/deploy/docker-compose/upgrade.md
Comment (Strum355) [inline doc/admin/deploy/docker-compose/upgrade.md:1]: Theres other commands up above in this file that arent prefixed with $, should they be?

Comment (efritz) [reply doc/admin/deploy/docker-compose/upgrade.md:1]: There's only one except and it's stuff like:

```sh
command 1
# expected result
command 2
# expected result
```

where the whole thing can be copied

Comment (Strum355) [reply doc/admin/deploy/docker-compose/upgrade.md:1]: So we do $ for single line ones, and no $ for multiple commands in one?

Comment (efritz) [reply doc/admin/deploy/docker-compose/upgrade.md:1]: yeah *unless* there's like real output

```sh
$ do a thing
some output here but it's not a bash comment
```

File: doc/admin/updates/automatic.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Automatic multi-version upgrades
+
+From **Sourcegraph 5.1 and later**, multi-version upgrades can be performed **automatically** as if they were a standard upgrade for the same deployment type. Automatic multi-version upgrades take the following general form:
+
+1. Determine if your instance is ready to Upgrade (check upgrade notes)
+2. Merge the latest Sourcegraph release into your deployment manifests
+3. With upstream changes to your manifests merged, start the new instance
```
[the remark this answers is not in the document]
Comment (Strum355) [reply doc/admin/updates/automatic.md:7]:
```suggestion
1. Determine if your instance is ready to Upgrade (check upgrade notes)
2. Merge the latest Sourcegraph release into your deployment manifests
3. Enable automatic upgrade for this upgrade by either:
	a. In the Updates section in site-admin (available on instances of Sourcegraph 5.1 and later), or
	b. set the `SRC_AUTOUPGRADE` environment variable to `true` on the migrator and frontend 5.1 or later deployment manifests (if the instance is of a version earlier than Sourcegraph 5.1)
4. With upstream changes to your manifests merged, start the new instance
```

File: doc/admin/updates/index.md [hunk 2]
```diff
@@ -26,16 +26,15 @@ To learn more about our release schedule see our [handbook](https://handbook.sou
...
 
 **Standard** 
 - Moves Sourcegraph one version forward (`v5.0.0` to `v5.1.0`), *this is usually one minor version unless the next version released is a major version*.
 - Requires minor downtime in deployments except kubernetes where rolling updates are possible.
 
 **Multiversion**
 - Moves Sourcegraph multiple versions forward (`v5.0.0` to `v5.2.0`).
-- Requires manual `migrator` operations
-- Requires downtime
+- Requires downtime while the database schemas and rewritten and unfinished out-of-band migrations are applied
```
Comment (Strum355) [inline doc/admin/updates/index.md:37]:
```suggestion
- Requires downtime while the database schemas are rewritten and unfinished out-of-band migrations are applied
```

File: doc/admin/updates/automatic.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+1. Detect that a previous version of Sourcegraph was/is currently running
+2. Plan the and persist the upgrade (which migrations to apply) based on the previously running version, the new target version, and the database state
+3. Start a new internal server that sends poison pills to disconnect old services from the databases and prevents new services from connecting before the upgrade completes
+4. Start a status server in place of the primary exposed port
+5. Runs the migration plan (performing the same steps as `migration upgrade ...` )
+6. Shuts down the internal and status servers and continues to boot normally
+
+## Viewing progress
+
+During an automatic multi-version upgrade, we'll attempt to boot a status serve in the frontend container that is running (or blocking) on an active upgrade attempt. If there is an upgrade failure that affects the frontend, this status page will not be available and the `frontend` container logs should be viewed. Optimistically, the status server will also be unreachable in the case that an upgrade performs quickly enough that there's no time for the status server to start.
```
Comment (Strum355) [inline doc/admin/updates/automatic.md:20]:
```suggestion
During an automatic multi-version upgrade, we'll attempt to boot a status server in the frontend container that is running (or blocking on) an active upgrade attempt. If there is an upgrade failure that affects the frontend, this status page will not be available and the `frontend` container logs should be viewed. Optimistically, the status server will also be unreachable in the case that an upgrade performs quickly enough that there's no time for the status server to start.
```

**Review by Strum355 — APPROVED** [review]

**Pull request merged**